File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-2.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    count = 0
    dfs(root, maxVal){
        if(!root){
            return null
        }
        if(root.val >= maxVal){
            maxVal = root.val
            this.count += 1
        }
        this.dfs(root.left, maxVal)
        this.dfs(root.right, maxVal)
    }
    goodNodes(root) {
        this.dfs(root, root.val)
        return this.count
    }
}

File: Data Structures & Algorithms/depth-of-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        maxVal = [0]
        def dfs(node, count):
            if(not(node)):
                return 0
            left = dfs(node.left, count + 1)
            right = dfs(node.right, count + 1)
            return 1 + max(left, right)

        return dfs(root, 1)
